vscode/src/commands/execute/doc.ts

```diff
@@ -104,7 +104,12 @@ export async function executeDocCommand(
             return undefined
         }
 
-        const { range, insertionPoint } = getDocumentableRange(editor)
+        let { range, insertionPoint } = getDocumentableRange(editor)
+
+        if (args?.range) {
+            range = args?.range
+            insertionPoint = args?.range.start
+        }
```

Comment:
```suggestion
        const { range, insertionPoint } = getDocumentableRange(editor)
```

Is this required? We should only use `getDocumentableRange` if possible. This will break for Python functions where we need to determine the insertion point depending on the syntax


https://github.com/sourcegraph/cody/assets/9516420/b65cd24e-8a36-41ef-9c1c-249408ff0cfb

Reply:
Yea good point! I've updated it to create a new editor with the document URL from args and let `getDocumentableRange` determine the correct range and insertionPoint